radix-engine-tests/tests/protocol/account_cuttlefish.rs

```diff
@@ -0,0 +1,209 @@
+    // Arrange
+    let mut ledger = LedgerSimulatorBuilder::new().build();
+    let (_, _, account) = ledger.new_account(false);
+
+    // Act
+    let receipt = ledger.execute_manifest(
+        ManifestBuilder::new()
+            .lock_fee_from_faucet()
+            .call_method(
+                account,
+                ACCOUNT_BALANCE_IDENT,
+                AccountBalanceInput {
+                    resource_address: XRD,
+                },
+            )
+            .build(),
+        vec![],
+    );
+
+    // Assert
+    let balance = receipt
+        .expect_commit_success()
+        .output::<AccountBalanceOutput>(1);
+    assert_eq!(balance, dec!(10_000));
+}
+
+#[test]
+fn balance_method_returns_zero_for_a_resource_the_account_doesnt_have() {
+    // Arrange
+    let mut ledger = LedgerSimulatorBuilder::new().build();
+    let (_, _, account) = ledger.new_account(false);
+
+    // Act
+    let receipt = ledger.execute_manifest(
+        ManifestBuilder::new()
+            .lock_fee_from_faucet()
+            .call_method(
+                account,
+                ACCOUNT_BALANCE_IDENT,
+                AccountBalanceInput {
+                    resource_address: ACCOUNT_OWNER_BADGE,
+                },
+            )
+            .build(),
+        vec![],
+    );
+
+    // Assert
+    let balance = receipt
+        .expect_commit_success()
+        .output::<AccountBalanceOutput>(1);
+    assert_eq!(balance, dec!(0));
+}
+
+#[test]
+fn non_fungible_local_ids_method_returns_an_empty_set_if_vault_doesnt_exist() {
+    // Arrange
+    let mut ledger = LedgerSimulatorBuilder::new().build();
+    let (_, _, account) = ledger.new_account(false);
+
+    // Act
+    let receipt = ledger.execute_manifest(
+        ManifestBuilder::new()
+            .lock_fee_from_faucet()
+            .call_method(
+                account,
+                ACCOUNT_NON_FUNGIBLE_LOCAL_IDS_IDENT,
+                AccountNonFungibleLocalIdsInput {
+                    resource_address: VALIDATOR_OWNER_BADGE,
+                    limit: 500,
+                },
+            )
+            .build(),
+        vec![],
+    );
+
+    // Assert
+    let ids = receipt
+        .expect_commit_success()
+        .output::<AccountNonFungibleLocalIdsOutput>(1);
+    assert_eq!(ids, indexset! {});
+}
+
+#[test]
+fn non_fungible_local_ids_method_returns_expected_set() {
+    // Arrange
+    let mut ledger = LedgerSimulatorBuilder::new().build();
+    let (_, _, account) = ledger.new_account(false);
+    let resource_address = ledger.create_non_fungible_resource(account);
+
+    // Act
+    let receipt = ledger.execute_manifest(
+        ManifestBuilder::new()
+            .lock_fee_from_faucet()
+            .call_method(
+                account,
+                ACCOUNT_NON_FUNGIBLE_LOCAL_IDS_IDENT,
+                AccountNonFungibleLocalIdsInput {
+                    resource_address,
+                    limit: 500,
```

Comment:
Hmm... not sure if this would actually test the account functionality though since the account just calls the vault's `Vault::non_fungible_local_ids` method so its just forwarding the call to that.

Reply:
Sure, but from a black box testing perspective, I think it's still worth covering some superficial tests on every method we add? It could be super simple where we just add 11 things to the account and test that asking for 10 gives us 10 :D.

Even for regression reasons. I could imagine a world where someone might refactor something and suddenly the limit isn't passed through or something.